# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/sync_policy.cc
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/sync_policy.cc
--- a/security/sandbox/chromium/sandbox/win/src/sync_policy.cc	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/sync_policy.cc	Wed Jun 12 11:10:48 2019 +0100
@@ -2,12 +2,12 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include "sandbox/win/src/sync_policy.h"
+
 #include <stdint.h>
 
 #include <string>
 
-#include "sandbox/win/src/sync_policy.h"
-
 #include "base/logging.h"
 #include "base/strings/stringprintf.h"
 #include "sandbox/win/src/ipc_tags.h"
@@ -26,38 +26,37 @@
 NTSTATUS ResolveSymbolicLink(const base::string16& directory_name,
                              const base::string16& name,
                              base::string16* target) {
-  NtOpenDirectoryObjectFunction NtOpenDirectoryObject = NULL;
+  NtOpenDirectoryObjectFunction NtOpenDirectoryObject = nullptr;
   ResolveNTFunctionPtr("NtOpenDirectoryObject", &NtOpenDirectoryObject);
 
-  NtQuerySymbolicLinkObjectFunction NtQuerySymbolicLinkObject = NULL;
-  ResolveNTFunctionPtr("NtQuerySymbolicLinkObject",
-                       &NtQuerySymbolicLinkObject);
+  NtQuerySymbolicLinkObjectFunction NtQuerySymbolicLinkObject = nullptr;
+  ResolveNTFunctionPtr("NtQuerySymbolicLinkObject", &NtQuerySymbolicLinkObject);
 
-  NtOpenSymbolicLinkObjectFunction NtOpenSymbolicLinkObject = NULL;
+  NtOpenSymbolicLinkObjectFunction NtOpenSymbolicLinkObject = nullptr;
   ResolveNTFunctionPtr("NtOpenSymbolicLinkObject", &NtOpenSymbolicLinkObject);
 
-  NtCloseFunction NtClose = NULL;
+  NtCloseFunction NtClose = nullptr;
   ResolveNTFunctionPtr("NtClose", &NtClose);
 
   OBJECT_ATTRIBUTES symbolic_link_directory_attributes = {};
   UNICODE_STRING symbolic_link_directory_string = {};
-  InitObjectAttribs(directory_name, OBJ_CASE_INSENSITIVE, NULL,
+  InitObjectAttribs(directory_name, OBJ_CASE_INSENSITIVE, nullptr,
                     &symbolic_link_directory_attributes,
-                    &symbolic_link_directory_string, NULL);
+                    &symbolic_link_directory_string, nullptr);
 
-  HANDLE symbolic_link_directory = NULL;
-  NTSTATUS status = NtOpenDirectoryObject(&symbolic_link_directory,
-                                          DIRECTORY_QUERY,
-                                          &symbolic_link_directory_attributes);
+  HANDLE symbolic_link_directory = nullptr;
+  NTSTATUS status =
+      NtOpenDirectoryObject(&symbolic_link_directory, DIRECTORY_QUERY,
+                            &symbolic_link_directory_attributes);
   if (!NT_SUCCESS(status))
     return status;
 
   OBJECT_ATTRIBUTES symbolic_link_attributes = {};
   UNICODE_STRING name_string = {};
   InitObjectAttribs(name, OBJ_CASE_INSENSITIVE, symbolic_link_directory,
-                    &symbolic_link_attributes, &name_string, NULL);
+                    &symbolic_link_attributes, &name_string, nullptr);
 
-  HANDLE symbolic_link = NULL;
+  HANDLE symbolic_link = nullptr;
   status = NtOpenSymbolicLinkObject(&symbolic_link, GENERIC_READ,
                                     &symbolic_link_attributes);
   CHECK(NT_SUCCESS(NtClose(symbolic_link_directory)));
@@ -66,8 +65,8 @@
 
   UNICODE_STRING target_path = {};
   unsigned long target_length = 0;
-  status = NtQuerySymbolicLinkObject(symbolic_link, &target_path,
-                                     &target_length);
+  status =
+      NtQuerySymbolicLinkObject(symbolic_link, &target_path, &target_length);
   if (status != STATUS_BUFFER_TOO_SMALL) {
     CHECK(NT_SUCCESS(NtClose(symbolic_link)));
     return status;
@@ -76,8 +75,8 @@
   target_path.Length = 0;
   target_path.MaximumLength = static_cast<USHORT>(target_length);
   target_path.Buffer = new wchar_t[target_path.MaximumLength + 1];
-  status = NtQuerySymbolicLinkObject(symbolic_link, &target_path,
-                                     &target_length);
+  status =
+      NtQuerySymbolicLinkObject(symbolic_link, &target_path, &target_length);
   if (NT_SUCCESS(status))
     target->assign(target_path.Buffer, target_length);
 
@@ -87,13 +86,13 @@
 }
 
 NTSTATUS GetBaseNamedObjectsDirectory(HANDLE* directory) {
-  static HANDLE base_named_objects_handle = NULL;
+  static HANDLE base_named_objects_handle = nullptr;
   if (base_named_objects_handle) {
     *directory = base_named_objects_handle;
     return STATUS_SUCCESS;
   }
 
-  NtOpenDirectoryObjectFunction NtOpenDirectoryObject = NULL;
+  NtOpenDirectoryObjectFunction NtOpenDirectoryObject = nullptr;
   ResolveNTFunctionPtr("NtOpenDirectoryObject", &NtOpenDirectoryObject);
 
   DWORD session_id = 0;
@@ -105,18 +104,16 @@
                                         base::StringPrintf(L"%d", session_id),
                                         &base_named_objects_path);
   if (!NT_SUCCESS(status)) {
-    DLOG(ERROR) << "Failed to resolve BaseNamedObjects path. Error: "
-                << status;
+    DLOG(ERROR) << "Failed to resolve BaseNamedObjects path. Error: " << status;
     return status;
   }
 
   UNICODE_STRING directory_name = {};
   OBJECT_ATTRIBUTES object_attributes = {};
-  InitObjectAttribs(base_named_objects_path, OBJ_CASE_INSENSITIVE, NULL,
-                    &object_attributes, &directory_name, NULL);
+  InitObjectAttribs(base_named_objects_path, OBJ_CASE_INSENSITIVE, nullptr,
+                    &object_attributes, &directory_name, nullptr);
   status = NtOpenDirectoryObject(&base_named_objects_handle,
-                                 DIRECTORY_ALL_ACCESS,
-                                 &object_attributes);
+                                 DIRECTORY_ALL_ACCESS, &object_attributes);
   if (NT_SUCCESS(status))
     *directory = base_named_objects_handle;
   return status;
@@ -174,7 +171,7 @@
                                        uint32_t event_type,
                                        uint32_t initial_state,
                                        HANDLE* handle) {
-  NtCreateEventFunction NtCreateEvent = NULL;
+  NtCreateEventFunction NtCreateEvent = nullptr;
   ResolveNTFunctionPtr("NtCreateEvent", &NtCreateEvent);
 
   // The only action supported is ASK_BROKER which means create the requested
@@ -182,7 +179,7 @@
   if (ASK_BROKER != eval_result)
     return false;
 
-  HANDLE object_directory = NULL;
+  HANDLE object_directory = nullptr;
   NTSTATUS status = GetBaseNamedObjectsDirectory(&object_directory);
   if (status != STATUS_SUCCESS)
     return status;
@@ -190,17 +187,17 @@
   UNICODE_STRING unicode_event_name = {};
   OBJECT_ATTRIBUTES object_attributes = {};
   InitObjectAttribs(event_name, OBJ_CASE_INSENSITIVE, object_directory,
-                    &object_attributes, &unicode_event_name, NULL);
+                    &object_attributes, &unicode_event_name, nullptr);
 
-  HANDLE local_handle = NULL;
+  HANDLE local_handle = nullptr;
   status = NtCreateEvent(&local_handle, EVENT_ALL_ACCESS, &object_attributes,
                          static_cast<EVENT_TYPE>(event_type),
                          static_cast<BOOLEAN>(initial_state != 0));
-  if (NULL == local_handle)
+  if (!local_handle)
     return status;
 
   if (!::DuplicateHandle(::GetCurrentProcess(), local_handle,
-                         client_info.process, handle, 0, FALSE,
+                         client_info.process, handle, 0, false,
                          DUPLICATE_CLOSE_SOURCE | DUPLICATE_SAME_ACCESS)) {
     return STATUS_ACCESS_DENIED;
   }
@@ -212,7 +209,7 @@
                                      const base::string16& event_name,
                                      uint32_t desired_access,
                                      HANDLE* handle) {
-  NtOpenEventFunction NtOpenEvent = NULL;
+  NtOpenEventFunction NtOpenEvent = nullptr;
   ResolveNTFunctionPtr("NtOpenEvent", &NtOpenEvent);
 
   // The only action supported is ASK_BROKER which means create the requested
@@ -220,7 +217,7 @@
   if (ASK_BROKER != eval_result)
     return false;
 
-  HANDLE object_directory = NULL;
+  HANDLE object_directory = nullptr;
   NTSTATUS status = GetBaseNamedObjectsDirectory(&object_directory);
   if (status != STATUS_SUCCESS)
     return status;
@@ -228,15 +225,15 @@
   UNICODE_STRING unicode_event_name = {};
   OBJECT_ATTRIBUTES object_attributes = {};
   InitObjectAttribs(event_name, OBJ_CASE_INSENSITIVE, object_directory,
-                    &object_attributes, &unicode_event_name, NULL);
+                    &object_attributes, &unicode_event_name, nullptr);
 
-  HANDLE local_handle = NULL;
+  HANDLE local_handle = nullptr;
   status = NtOpenEvent(&local_handle, desired_access, &object_attributes);
-  if (NULL == local_handle)
+  if (!local_handle)
     return status;
 
   if (!::DuplicateHandle(::GetCurrentProcess(), local_handle,
-                         client_info.process, handle, 0, FALSE,
+                         client_info.process, handle, 0, false,
                          DUPLICATE_CLOSE_SOURCE | DUPLICATE_SAME_ACCESS)) {
     return STATUS_ACCESS_DENIED;
   }